packages/server/src/cds/routes.ts

```diff
@@ -0,0 +1,82 @@
+// SPDX-FileCopyrightText: Copyright Orangebot, Inc. and Medplum contributors
+// SPDX-License-Identifier: Apache-2.0
+import { allOk, badRequest, ContentType, getStatus, Operator } from '@medplum/core';
+import type { Bot } from '@medplum/fhirtypes';
+import type { Request, Response } from 'express';
+import { Router } from 'express';
+import { executeBot } from '../bots/execute';
+import { getBotDefaultHeaders, getResponseBodyFromResult } from '../bots/utils';
+import { getAuthenticatedContext } from '../context';
+import { authenticateRequest } from '../oauth/middleware';
+
+// CDS Hooks
+// https://cds-hooks.hl7.org/
+
+export const cdsRouter = Router().use(authenticateRequest);
+
+// Discovery: https://cds-hooks.hl7.org/#discovery
+cdsRouter.get('/', async (_req: Request, res: Response) => {
+  const { repo } = getAuthenticatedContext();
+  repo.setExtendedMode(true);
+
+  // The CDS Hooks spec does not define pagination for discovery
+  // Most servers will have under 10 CDS services, so we set a high max count.
+  const maxCount = 100;
+
+  const bots = await repo.searchResources<Bot>({
+    resourceType: 'Bot',
+    filters: [{ code: 'cds-hook', operator: Operator.MISSING, value: 'false' }],
+    count: maxCount,
+  });
```

Comment:
If someone does add > 100 CDS bots, it seems like this will find some subset of the bots available to list. Should we fail loudly in that scenario? I'm imagining something roughly like:

```ts
if (bots.length === maxCount) {
  throw new Error('CDS bot limit exceeded`);
}
```

Reply:
Yeah, there isn't a great answer here.  It's a weird oversight in the spec to not allow pagination.

I need to noodle on throwing an error.  I see the merit in failing loudly, but it also feels risky.

We could just allow for sorting and pagination using the FHIR query params (i.e., `_sort`, `_offset`, etc)